test/unit/snappy.test.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const mock = require('../tools/mock');
+const snappy = optionalRequire('snappy');
+
+function optionalRequire(mod) {
+  try {
+    return require(mod);
+  } catch {
+    return false;
+  }
+}
+
+describe('Compression', function () {
+  let client;
+  /** @type {mock.MockServer} */
+  let server;
+  before(async function () {
+    if (!snappy) this.skip();
+    server = await mock.createServer();
+    client = this.configuration.newClient(`mongodb://${server.uri()}`, { compressors: 'snappy' });
+  });
+
+  describe('Snappy', () => {
+    it('should compress messages sent with snappy', async () => {
+      server.setMessageHandler(request => {
+        const doc = request.document;
+        if (doc.ismaster || doc.hello) {
+          return request.reply({ ...mock.DEFAULT_ISMASTER, compression: ['snappy'] });
+        }
+        if (doc.insert === 'snappy') {
+          return request.reply({ ok: 1 });
+        }
+      });
+      // The mock server uses snappy to decode messages so
+      // if this passes we implicitly test snappy is working
+      // TODO(NODE-XXXX): Add more comprehensive round trip testing
```

Comment:
This is a test that does nothing special essentially, the mock server uses snappy 6 and doesn't give us hooks where we could catch and run the buffer through snappy ourselves at this layer. I think there is certainly more improvement but at least this prevents regression (as long as the mock server doesn't change). Open to more ideas about assertions.